File: module-1/11_Inheritance/exercise-student/java/src/main/java/com/techelevator/CheckingAccount.java
package com.techelevator;

public class CheckingAccount extends BankAccount {

    public CheckingAccount(String accountHolderName, String accountNumber) {
        super(accountHolderName, accountNumber);
    }


    public CheckingAccount(String accountHolderName, String accountNumber, int balance) {
        super(accountHolderName, accountNumber, balance);
    }


    @Override

    public int withdraw(int amountToWithdraw) {
        //if balance < 0
        //$10.00 overdraft fee
        //then  overdraft up to $100
        //if overdraft > 100 then transfer fails balance = balance


        if (super.getBalance() >= amountToWithdraw) {
            super.withdraw(amountToWithdraw);
        }
        else {
            if (super.getBalance() - amountToWithdraw >= -100) {
                super.withdraw(amountToWithdraw + 10);
            }
        }

        return getBalance();
    }
}

File: module-1/12_Polymorphism/exercise-student/dotnet/BankTellerExercise/CheckingAccount.cs
namespace BankTellerExercise
{
    public class CheckingAccount : BankAccount
    {

        public CheckingAccount(string accountHolder, string accountNumber) : base(accountHolder, accountNumber)
        {

        }

        public CheckingAccount(string accountHolder, string accountNumber, int balance) : base(accountHolder, accountNumber, balance)
        {

        }

        public override int Withdraw(int amountToWithdraw)
        {
            // Only allow the withdraw if the balance isn't going to go below -$100
            if (Balance - amountToWithdraw > -100)
            {
                // Withdraw the $$
                base.Withdraw(amountToWithdraw);
                // If the balance dips below 0, assess $10 charge
                if (Balance < 0)
                {
                    base.Withdraw(10);
                }
            }
            return Balance;
        }
    }
}
